### PR TITLE
[FLINK-32848][tests][JUnit5 migration] Migrate flink-runtime/shuffle tests to JUnit5

### PR DESCRIPTION
**Pull request by X-czh**
## What is the purpose of the change

Migrate flink-runtime/shuffle tests to JUnit5.

## Verifying this change

The change itself is for migrating UT.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-runtime/src/test/java/org/apache/flink/runtime/shuffle/ShuffleMasterTest.java [hunk 1]
```diff
@@ -79,16 +75,16 @@ public void testShuffleMasterLifeCycle() throws Exception {
                     EXTERNAL_PARTITION_RELEASE_EVENT,
                     EXTERNAL_PARTITION_RELEASE_EVENT,
                 };
-        assertArrayEquals(expectedPartitionEvents, TestShuffleMaster.partitionEvents.toArray());
+        assertThat(TestShuffleMaster.partitionEvents.toArray()).isEqualTo(expectedPartitionEvents);
```
Comment (ferenc-csaky) [inline flink-runtime/src/test/java/org/apache/flink/runtime/shuffle/ShuffleMasterTest.java:78]: Possible Assert4j shorthand:
```java
        assertThat(TestShuffleMaster.partitionEvents)
                .containsExactly(
                        PARTITION_REGISTRATION_EVENT,
                        PARTITION_REGISTRATION_EVENT,
                        EXTERNAL_PARTITION_RELEASE_EVENT,
                        EXTERNAL_PARTITION_RELEASE_EVENT);
```

File: flink-runtime/src/test/java/org/apache/flink/runtime/shuffle/ShuffleMasterTest.java [hunk 2]
```diff
@@ -99,7 +95,7 @@ public void testStopTrackingPartition() throws Exception {
...
```
Comment (ferenc-csaky) [inline flink-runtime/src/test/java/org/apache/flink/runtime/shuffle/ShuffleMasterTest.java:98]: Possible Assert4j shorthand:
```java
        assertThat(TestShuffleMaster.partitionEvents)
                .containsExactly(
                        PARTITION_REGISTRATION_EVENT,
                        PARTITION_REGISTRATION_EVENT,
                        PARTITION_REGISTRATION_EVENT,
                        PARTITION_REGISTRATION_EVENT,
                        EXTERNAL_PARTITION_RELEASE_EVENT,
                        EXTERNAL_PARTITION_RELEASE_EVENT);
```

**Review by ferenc-csaky — COMMENTED** [review]
Thanks for making these chamges, added 2 improvement suggestions.

File: flink-runtime/src/test/java/org/apache/flink/runtime/shuffle/ShuffleMasterTest.java (hunk 2 above)
Comment (X-czh) [reply flink-runtime/src/test/java/org/apache/flink/runtime/shuffle/ShuffleMasterTest.java]: Thanks for pointing that, I've prepared an improvement commit

**Review by ferenc-csaky — APPROVED** [review]
LGTM!

**Comment by X-czh** [comment]
Hi @1996fanrui, could you review the change when you are free, much appreciated~

**Comment by X-czh** [comment]
Hi @FangYongs, could you help check the change when you are free?

**Comment by FangYongs** [comment]
Thanks @X-czh for contributions, LGTM

**Pull request merged**